arwen/contexts/storage.go

```diff
@@ -63,9 +68,11 @@ func (context *storageContext) GetStorage(key []byte) []byte {
 	}
 
 	value, _ := context.blockChainHook.GetStorageData(context.address, key)
-	storageUpdates[string(key)] = &vmcommon.StorageUpdate{
-		Offset: key,
-		Data:   value,
+	if value != nil {
```

Comment:
No, this simply brings a value from the Trie into the `StorageUpdates`, which acts as a RW cache. If a value is brought from the `BlockchainHook` (Trie) just for reading, it will stay unmodified in `StorageUpdates`. 

If an entry is deleted, it is added in `StorageUpdates` with `[]byte{}`. See `storageContext.SetStorage()`